src/ui/blocks/SineSource.hpp

```diff
@@ -40,31 +39,35 @@ struct SineSource : public gr::Block<SineSource<T>, gr::BlockingIO<true>> {
     }
 
     ~SineSource() {
+        std::unique_lock guard(mutex);
```

Comment:
Should be replaced and synchronised with [EPIC: List of GR 3.10 blocks & Others to be ported to GR 4.0](https://github.com/fair-acc/graph-prototype/issues/161). Also there is already a sine and cosine source in GR 4.0. 

@drslebedev maybe you could comment/document the usage and check this if this conforms to the functionality in GR 3.X.

Reply:
Here is the example of `SignalGenerator` which can generate sine wave (this is the default signal type), note that `ClockSource` is required sa input to `SignalGenerator`:
```cpp
constexpr std::uint32_t n_samples   = 200;
constexpr float         sample_rate = 1000.f;
Graph                   testGraph;
auto                   &clockSrc  = testGraph.emplaceBlock<gr::basic::ClockSource<float>>({ { "sample_rate", sample_rate }, { "n_samples_max", n_samples }, { "name", "ClockSource" } });
auto                   &signalGen = testGraph.emplaceBlock<SignalGenerator<float>>({ { "sample_rate", sample_rate }, { "name", "SignalGenerator" } });

testGraph.connect<"out">(clockSrc).to<"in">(signalGen);
testGraph.connect<"out">(signalGen).to<"in">(sink);

scheduler::Simple sched{ std::move(testGraph) };
sched.runAndWait();
```